blueprints/app/files/.stylelintignore

```diff
@@ -0,0 +1,25 @@
+# unconventional js
```

Comment:
This file is currently an exact match of `.prettierignore`.

Reply:
This should be sufficient, in my experience. The stylelintignore in our app at work is very similar.

Reply:
Why does it need to be an exact match? Some of these aren't applicable to .css files I think? Adding them just to be consistent seems a bit strange to me. It looks like a mistake when reading the file 😄.

Reply:
That's a fair point. But I'm unclear which to omit. Most of these folders could have various types of files in them, and we might as well just tell stylelint to ignore searching them regardless.

Reply:
That's a fair point. But I'm unclear which to omit. Most of these folders could have various types of files in them, and we might as well just tell stylelint to ignore searching them regardless.

Reply:
We run the same setup at work for projects and all we really need at the moment is:

```
/dist/
```

I guess for addons, it might be slightly different.

Reply:
Yeah, I imagine most addons could get away with a small/minimal list, but the same could be said to some extent for the JS/prettier ignore lists. Since those lists have been seemingly built to include everything conceivable, it seems like we might as well use the same list. But if we know of some items that cannot possibly have CSS files under them, then I'm open to removing them.

Reply:
I made https://github.com/ember-cli/ember-cli/issues/10063 a while ago, but haven't got to it yet.
I think a lot of entries aren't relevant anymore for new(er) apps.
Personally, I don't see why we should include entries that are not needed.

Reply:
I went ahead and removed:
* `bower`-related items - I suspect it would be very rare for anyone to use that with a new app
* `.lint-todo/`- not related (from ember-template-lint) 
* `.eslintcache` - not related
* `coverage/` - not related

Kept:
* `blueprints/` - if someone creates their own Ember CLI [blueprints](https://cli.emberjs.com/release/advanced-use/blueprints/)
* `dist/` - common for various build tool output
* `tmp/` - common for temp files
* `vendor/` - this is a common folder used for manually storing third-party deps, even if it doesn't come with Ember apps by default
* `node_modules/` - obviously we want to ignore this, I don't see it mentioned as a default value in the [doc](https://stylelint.io/user-guide/ignore-code/#files-entirely) but this might not be documented, also note that default values could change
* `.node_modules.ember-try/` - common dependency in Ember apps

My general approach here is to ensure that folders from common dependencies (e.g. ember-try) or folders that are common in the frontend ecosystem (e.g. `vendor`) get ignored, with the hope that the vast majority of users won't have to worry about this / touch this file.

Reply:
@bmish please keep `.lint-todo/`, that is not a technology unique to ember-template-lint. TODO infrastructure exists for ember-template-lint, eslint, and stylelint.

Reply:
More info here: https://github.com/lint-todo/stylelint-formatter-todo

Reply:
Thanks! Just added back.

Reply:
- The `tmp` folder in Ember projects was only a thing before the system temp folder was used I think, so I don't think it's needed for Ember apps that will get this update
- `vendor` is not included by default anymore in the blueprint as it's an outdated concept with tools like `ember-auto-import`, so to me it makes sense to omit it here as well
- `node_modules` is ignored by default, it's mentioned here: https://stylelint.io/user-guide/configure/#ignorefiles
- `.lint-todo` doesn't contain any css files, so I don't see why it should be added?

Reply:
I removed `node_modules`, thanks for confirming that.

I understand `tmp` and `vendor` aren't used by default anymore, but I still think they're useful to include. Many large/complex/older Ember apps I work with still use them, both for custom and legacy reasons. 

- `tmp` is commonly used for experimentation / temporary / custom script output 
- `vendor` is a common pattern in the web world for storing third-party resources, even if not necessary in Ember anymore as a result of ember-auto-import, and it was only just removed from Ember

I think there's also something to be said for keeping these ignore lists reasonably consistent between JS/prettier/stylelint. Seeing different behavior from each tool because their ignore lists are inconsistent could cause confusion.

Reply:
Older apps won't be affected by this afaict. If you update to the version this ships in, you won't be using the `tmp` folder anymore. If it's for custom reasons, I feel that's not really Ember CLI's responsibility. I have no projects where these folders are used. I also don't think these ignore files should be consistent for the sake of consistency, that's how they got bloated in the first place I think, and how people might become confused about what's actually needed and what's not. The app blueprint is also used for creating new projects, which means new projects start off with things in these files that are no longer relevant.

Reply:
Okay, I have removed `tmp` and `vendor`.

Reply:
Does it look good now? We should merge soon.

Reply:
`.lint-todo` doesn't contain any css files afaik, so I think we can safely remove that one as well.

Reply:
I removed `.lint-todo` again. If CSS files are expected in that folder, let us know.

Also removed `!.*` since I don't think hidden CSS files are common.